packages/expo-modules-core/ios/Swift/Exceptions/Exception.swift

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+open class Exception: CodedError, ChainableException, CustomStringConvertible, CustomDebugStringConvertible {
+  open var name: String {
+    return String(describing: Self.self)
+  }
+
+  /**
+   String describing the reason of the exception.
+   */
+  open var reason: String {
+    "undefined reason"
+  }
+
+  /**
+   The location in code where the exception was created.
+   */
+  open var location: ExceptionLocation
```

Comment:
I'm looking for the best work to describe this 👀 
I know that you've discarded `source`, but `location` seems a bit odd as well (I understand `location` as a physical presence at some geographic area/point)
```suggestion
  open var origin: ExceptionOrigin
```